core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -451,6 +542,58 @@ private ExprValue exprDate(ExprValue exprValue) {
     }
   }
 
+  /**
+   * DateTime implementation for ExprValue.
+   *
+   * @param dateTime ExprValue of String type.
+   * @param timeZone ExprValue of String type.
+   * @return ExprValue of date type.
+   */
+  private ExprValue exprDateTime(ExprValue dateTime, ExprValue timeZone) {
+    String defaultTimeZone = TimeZone.getDefault().getID();
+    DateTimeFormatter formatDT = DateTimeFormatter.ofPattern("uuuu-MM-dd HH:mm:ss[xxx]")
+        .withResolverStyle(ResolverStyle.STRICT);
+
+    try {
+      LocalDateTime ldtFormatted = LocalDateTime.parse(dateTime.stringValue(), formatDT);
+      if (timeZone.isNull()) {
+        return new ExprDatetimeValue(ldtFormatted);
+      }
+    } catch (DateTimeParseException e) {
+      return ExprNullValue.of();
+    }
+    // Used by exprDateTimeNoTZ function
+
+    ExprValue convertTZResult;
+
+    try {
+      ZonedDateTime zdtWithZoneOffset = ZonedDateTime.parse(dateTime.stringValue(), formatDT);
+      ZoneId fromTZ = zdtWithZoneOffset.getZone();
+
+      convertTZResult = exprConvertTZ(
+          new ExprDatetimeValue(zdtWithZoneOffset.toLocalDateTime()),
+          new ExprStringValue(String.valueOf(fromTZ)),
+          new ExprStringValue(String.valueOf(ZoneId.of(timeZone.stringValue()))));
+    } catch (DateTimeParseException e) {
+      LocalDateTime ldtFormatted = LocalDateTime.parse(dateTime.stringValue(), formatDT);
+      convertTZResult = exprConvertTZ(
```

Comment:
Can you extract call to `exprConvertTZ` outside of try-catch block?

Reply:
Resolved.